### PR TITLE
Random misc world gen mappings

### PR DESCRIPTION
**Pull request by jaskarth**
Here's a bunch of new names, refactors, and a splash of javadoc. Please let me know if there are any issues.

### CODE REVIEW

File: mappings/net/minecraft/world/biome/layer/BiomeLayers.mapping [hunk 1]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_3645 net/minecraft/world/biome/layer/BiomeLayers
+	FIELD field_26709 CATAGORY_MAP Lit/unimi/dsi/fastutil/ints/Int2IntMap;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/layer/BiomeLayers.mapping]:
```suggestion
	FIELD field_26709 BY_CATEGORY Lit/unimi/dsi/fastutil/ints/Int2IntMap;
```

File: mappings/net/minecraft/world/biome/layer/BiomeLayers.mapping [hunk 2]
```diff
@@ -24,4 +25,10 @@ CLASS net/minecraft/class_3645 net/minecraft/world/biome/layer/BiomeLayers
 		ARG 5 contextProvider
 	METHOD method_15849 (JJ)Lnet/minecraft/class_3631;
 		ARG 2 salt
+	METHOD method_31116 (Lit/unimi/dsi/fastutil/ints/Int2IntOpenHashMap;)V
+		ARG 0 map
+	METHOD method_31117 putCatagory (Lit/unimi/dsi/fastutil/ints/Int2IntOpenHashMap;Lnet/minecraft/class_3645$class_5503;I)V
+		ARG 0 map
+		ARG 1 catagory
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/layer/BiomeLayers.mapping]:
```suggestion
	METHOD method_31117 putCategory (Lit/unimi/dsi/fastutil/ints/Int2IntOpenHashMap;Lnet/minecraft/class_3645$class_5503;I)V
		ARG 0 map
		ARG 1 category
```

File: mappings/net/minecraft/world/biome/layer/ScaleLayer.mapping [hunk 3]
```diff
@@ -1,3 +1,7 @@
 CLASS net/minecraft/class_3656 net/minecraft/world/biome/layer/ScaleLayer
 	METHOD method_15853 sample (Lnet/minecraft/class_3628;IIII)I
 		ARG 1 context
+		ARG 2 tl
+		ARG 3 tr
+		ARG 4 bl
+		ARG 5 br
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/layer/ScaleLayer.mapping:7]: Back to intermediary v1! (joke apart, I think javadocs would be really appreciated to explain these params)

File: mappings/net/minecraft/world/biome/layer/util/LayerSampleContext.mapping [hunk 4]
```diff
@@ -1,6 +1,8 @@
 CLASS net/minecraft/class_3628 net/minecraft/world/biome/layer/util/LayerSampleContext
 	METHOD method_15828 createSampler (Lnet/minecraft/class_4;Lnet/minecraft/class_3625;Lnet/minecraft/class_3625;)Lnet/minecraft/class_3625;
 		ARG 1 operator
+		ARG 2 parent1
+		ARG 3 parent2
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/layer/util/LayerSampleContext.mapping]:
```suggestion
		ARG 2 firstParent
		ARG 3 secondParent
```

File: mappings/net/minecraft/world/biome/source/BiomeLayerSampler.mapping [hunk 5]
```diff
@@ -4,3 +4,6 @@ CLASS net/minecraft/class_3642 net/minecraft/world/biome/source/BiomeLayerSample
 	METHOD <init> (Lnet/minecraft/class_3627;)V
 		ARG 1 layerFactory
 	METHOD method_16341 sample (Lnet/minecraft/class_2378;II)Lnet/minecraft/class_1959;
+		ARG 1 registry
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/source/BiomeLayerSampler.mapping]:
```suggestion
		ARG 1 biomeRegistry
```

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping [hunk 6]
```diff
@@ -8,3 +10,19 @@ CLASS net/minecraft/class_4953 net/minecraft/world/gen/feature/TwistingVinesFeat
...
+	METHOD method_26265 tryGenerateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;III)Z
+		ARG 0 world
+		ARG 1 random
+		ARG 2 pos
+		ARG 3 horizontalSpread
+		ARG 4 verticalSpread
+		ARG 5 length
+	METHOD method_27220 canGenerate (Lnet/minecraft/class_1936;Lnet/minecraft/class_2338$class_2339;)Z
+		ARG 0 world
+		ARG 1 mutable
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]:
```suggestion
		ARG 1 pos
```

File: mappings/net/minecraft/world/gen/feature/size/FeatureSize.mapping [hunk 7]
```diff
@@ -1,8 +1,13 @@
 CLASS net/minecraft/class_5201 net/minecraft/world/gen/feature/size/FeatureSize
+	COMMENT In charge of determining the amount of space needed by a tree to generate.
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/size/FeatureSize.mapping:2]: Is it only used by tree features?

File: mappings/net/minecraft/world/gen/feature/size/FeatureSize.mapping [hunk 8]
```diff
@@ -1,8 +1,13 @@
...
 	FIELD field_24146 minClippedHeight Ljava/util/OptionalInt;
 	FIELD field_24922 TYPE_CODEC Lcom/mojang/serialization/Codec;
 	METHOD <init> (Ljava/util/OptionalInt;)V
 		ARG 1 minClippedHeight
 	METHOD method_27377 getMinClippedHeight ()Ljava/util/OptionalInt;
-	METHOD method_28820 createCodecBuilder ()Lcom/mojang/serialization/codecs/RecordCodecBuilder;
+	METHOD method_27378 getRadius (II)I
+		COMMENT The radius that the tree needs to be empty or replacable in order for it to generate.
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/size/FeatureSize.mapping]:
```suggestion
		COMMENT The radius that the tree needs to be empty or replaceable in order for it to generate.
```
I don't understand this sentence

File: mappings/net/minecraft/world/gen/feature/size/FeatureSize.mapping (hunk 7 above)
Comment (jaskarth) [reply mappings/net/minecraft/world/gen/feature/size/FeatureSize.mapping:2]: Yeah, it's only used by trees at this time.

File: mappings/net/minecraft/world/biome/layer/ScaleLayer.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/world/biome/layer/ScaleLayer.mapping:7]: Looks like top/bottom left/right

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]
Needs to target 20w51a

**Pull request merged**